### PR TITLE
refac: Simplify create_time_series method in settlement report job

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [x] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py [hunk 1]
```diff
@@ -213,78 +174,26 @@ def _generate_time_series(
...
-        for i in range(desired_quantity_column_count)
-    ]
+        *quantity_column_names,
+    )
 
-    df_with_array_column = (
-        filtered_metering_points.withColumn(
-            EphemeralColumns.start_of_day,
+
+def _add_start_of_day_column(
```
Comment (jpm-halfspace) [inline source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py:181]: This should take the fields as arguments. utils are by definition ment to be shared and should therefore not have any hardcoded values

File: source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py [hunk 2]
```diff
@@ -198,11 +126,44 @@ def _read_and_filter_from_view(
...
 )
 def _generate_time_series(
-    filtered_metering_points: DataFrame,
-    desired_quantity_column_count: int,
+    filtered_time_series_points: DataFrame,
+    desired_number_of_quantity_columns: int,
     time_zone: str,
 ) -> DataFrame:
-    _result_columns = [
+    filtered_time_series_points = _add_start_of_day_column(
```
Comment (jpm-halfspace) [inline source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py:133]: I find this a little confusing. I would probably have either made it a column function, i.e., a function that takes a `Column` and returns a `Column` or just do this here. Abstracting this, IMO, hinders the readability of the function

**Review by jpm-halfspace — APPROVED** [review]
I have minor suggestions, but overall I like this way of doing it.

File: source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py (hunk 1 above)
Comment (jpm-halfspace) [reply source/databricks/settlement_report/settlement_report_job/domain/time_series_factory.py:181]: something like this perhaps

```py
def get_start_of_day(col: Column | str, time_zone: str = "Europe/Copenhagen") -> Column:
    col = F.col(col) if isinstance(col, str) else col
    return F.to_utc_timestamp(
        F.date_trunc("DAY", F.from_utc_timestamp(col, time_zone), time_zone)
    )
```

Then you can use it as
```py
df = df.withColumn("start_of_day", get_start_of_day("observation_time"))
```

**Review by johevemi — APPROVED** [review]

**Pull request merged**